cartographer/mapping/internal/2d/scan_matching/real_time_correlative_scan_matcher_2d_test.cc

```diff
@@ -52,75 +108,82 @@ class RealTimeCorrelativeScanMatcherTest : public ::testing::Test {
-    point_cloud_.emplace_back(-0.125f, 0.075f, 0.f);
-    point_cloud_.emplace_back(-0.125f, 0.025f, 0.f);
     range_data_inserter_->Insert(
         sensor::RangeData{Eigen::Vector3f::Zero(), point_cloud_, {}},
-        &probability_grid_);
-    probability_grid_.FinishUpdate();
-    {
-      auto parameter_dictionary = common::MakeDictionary(
-          "return {"
-          "linear_search_window = 0.6, "
-          "angular_search_window = 0.16, "
-          "translation_delta_cost_weight = 0., "
-          "rotation_delta_cost_weight = 0., "
-          "}");
-      real_time_correlative_scan_matcher_ =
-          absl::make_unique<RealTimeCorrelativeScanMatcher2D>(
-              CreateRealTimeCorrelativeScanMatcherOptions(
-                  parameter_dictionary.get()));
-    }
+        grid_.get());
+    grid_->FinishUpdate();
   }
 
   ValueConversionTables conversion_tables_;
-  ProbabilityGrid probability_grid_;
-  std::unique_ptr<ProbabilityGridRangeDataInserter2D> range_data_inserter_;
+  std::unique_ptr<Grid2D> grid_;
+  std::unique_ptr<RangeDataInserterInterface> range_data_inserter_;
   sensor::PointCloud point_cloud_;
   std::unique_ptr<RealTimeCorrelativeScanMatcher2D>
       real_time_correlative_scan_matcher_;
 };
 
-TEST_F(RealTimeCorrelativeScanMatcherTest,
+INSTANTIATE_TEST_CASE_P(
+    RealTimeCorrelativeScanMatcherTest, RealTimeCorrelativeScanMatcherTest,
+    ::testing::Values(::cartographer::mapping::GridType::PROBABILITY_GRID,
+                      ::cartographer::mapping::GridType::TSDF));
+
+TEST_P(RealTimeCorrelativeScanMatcherTest,
        ScorePerfectHighResolutionCandidate) {
+  if (GetParam() == ::cartographer::mapping::GridType::PROBABILITY_GRID) {
+    SetUpProbabilityGrid();
+  } else if (GetParam() == ::cartographer::mapping::GridType::TSDF) {
+    SetUpTSDF();
+  }
   const std::vector<sensor::PointCloud> scans =
       GenerateRotatedScans(point_cloud_, SearchParameters(0, 0, 0., 0.));
-  const std::vector<DiscreteScan2D> discrete_scans = DiscretizeScans(
-      probability_grid_.limits(), scans, Eigen::Translation2f::Identity());
+  const std::vector<DiscreteScan2D> discrete_scans =
+      DiscretizeScans(grid_->limits(), scans, Eigen::Translation2f::Identity());
   std::vector<Candidate2D> candidates;
   candidates.emplace_back(0, 0, 0, SearchParameters(0, 0, 0., 0.));
   real_time_correlative_scan_matcher_->ScoreCandidates(
-      probability_grid_, discrete_scans, SearchParameters(0, 0, 0., 0.),
+      *grid_.get(), discrete_scans, SearchParameters(0, 0, 0., 0.),
       &candidates);
   EXPECT_EQ(0, candidates[0].scan_index);
   EXPECT_EQ(0, candidates[0].x_index_offset);
   EXPECT_EQ(0, candidates[0].y_index_offset);
   // Every point should align perfectly.
-  EXPECT_NEAR(0.7, candidates[0].score, 1e-2);
+  if (GetParam() == ::cartographer::mapping::GridType::PROBABILITY_GRID) {
+    EXPECT_NEAR(0.7, candidates[0].score, 1e-2);
+  } else if (GetParam() == ::cartographer::mapping::GridType::TSDF) {
```

Comment:
Since these tests share very little code and have different expectations, I think non-parametric tests (calling some same functions to generate the options and point cloud) would be more readable. Sorry for the double work.